docs/development/testing/cmek.md

```diff
@@ -52,18 +61,112 @@
+            dataVolumes:
+               - name: alpha
+                  type: pd-balanced
+                  size: 25Gi
+               - name: beta
+                  type: SCRATCH
+                  size: 25Gi
+            zones:
+               - europe-west1-c
+            systemComponents:
+               allow: true
+            providerConfig:
+               apiVersion: gcp.provider.extensions.gardener.cloud/v1alpha1
+               kind: WorkerConfig
+               volume:
+                  interface: "SCSI" # applies only for scratch disks. misleading field name
+                  encryption:
+                  kmsKeyName: "projects/projectId/locations/<zoneName>/keyRings/<keyRingName>/cryptoKeys/alpha"
+                  kmsKeyServiceAccount: "user@projectId.iam.gserviceaccount.com" 
+      ```
+1. Apply the shoot YAML agaisnt the `nodeless` cluster and wait till once the shoot has reconciled and the machine(s) in pool are in `Running` phase.
+1. Use `gardenctl` to target the shooted seed and get the provider machine name and zone
+   ```
+   kubectl get mc -A -o=jsonpath='{.items[*].spec.providerID}'
+   gce:///projectId/zone/shoot--userid--shootName-worker-1-z1-65c6b-9ptm7
+   ```
+1. You can check that the encrypted disks have been created:
+    1. Describe the instance: 
+    `gcloud compute instances describe shoot--userid--shootName-worker-1-z1-65c6b-9ptm7 --zone zone` 
+    1. The persistent disks should look like below with  have a disk encryoption key
+    ```yaml
+      disks:
+      - autoDelete: true
+      boot: true
+      deviceName: persistent-disk-0
+      diskEncryptionKey:
+         kmsKeyName: projects/projectId/locations/<zoneName>/keyRings/<keyRingName>/cryptoKeys/alpha"
+         kmsKeyServiceAccount: user@projectId.iam.gserviceaccount.com 
+      diskSizeGb: '50'
+      index: 0
+      interface: SCSI
+      kind: compute#attachedDisk
+      mode: READ_WRITE
+      source: <...>
+      type: PERSISTENT
+      - autoDelete: true
+      boot: false
+      deviceName: local-ssd-0
+      diskSizeGb: '375'
+      index: 1
+      interface: SCSI
+      kind: compute#attachedDisk
+      mode: READ_WRITE
+      savedState: DISK_SAVED_STATE_UNSPECIFIED
+      type: SCRATCH
+      - autoDelete: true
+      boot: false
+      deviceName: persistent-disk-2
+      diskEncryptionKey:
+         kmsKeyName: projects/projectId/locations/<zoneName>/keyRings/<keyRingName>/cryptoKeys/alpha
+         kmsKeyServiceAccount: user@projectId.iam.gserviceaccount.com 
+      diskSizeGb: '25'
+      index: 2
+      interface: SCSI
+      kind: compute#attachedDisk
+      mode: READ_WRITE
+      source: <...>
+      type: PERSISTENT
+    ```
+1. One can change the shoot YAML by adding removing volumes and once the shoot has re-conciled and the new machine in `Running` phase, the persistent disks associated wth the machine VM should be correspondingly added/removed and still be configured with the specified encryoption key.
```

Comment:
One can change the shoot YAML by adding/removing volumes. Once the shoot has reconciled with the new machines in the `Running` phase, the persistent disks associated with the VM should be added/removed and still be configured with the specified encryption key.

Reply:
done